src/include/daos/agent.h

```diff
@@ -27,14 +27,40 @@
 #include <daos/drpc.h>
 
 /**
- * Default Unix Domain Socket path for the DAOS agent dRPC connection
+ *  Called during libray initialization to craft socket path for agent.
  */
-#define DEFAULT_DAOS_AGENT_DRPC_SOCK "/var/run/daos_agent/agent.sock"
+int dc_agent_init(void);
+
+/**
+ *  Called during library finalization to free allocated agent resources
+ */
+void dc_agent_fini(void);
+
+/**
+ * Path to be used to communicate with the DAOS Agent set at library init.
+ */
+extern char *dc_agent_sockpath;
+
+/**
+ * Default runtime directory for daos_agent
+ */
+#define DAOS_AGENT_DRPC_DIR "/var/run/daos_agent/"
 
 /**
  * Environment variable for specifying an alternate dRPC socket path
  */
-#define DAOS_AGENT_DRPC_SOCK_ENV "DAOS_AGENT_DRPC_SOCK"
+#define DAOS_AGENT_DRPC_DIR_ENV "DAOS_AGENT_DRPC_DIR"
+
+/**
+ * Socket name used to craft path from environment variable
+ */
+#define DAOS_AGENT_DRPC_SOCK_NAME "agent.sock"
+
+/**
+ * Default Unix Domain Socket path for the DAOS agent dRPC connection
+ */
+#define DEFAULT_DAOS_AGENT_DRPC_SOCK	(DAOS_AGENT_DRPC_DIR \
+					DAOS_AGENT_DRPC_SOCK_NAME)
```

Comment:
Are these macros exported for the tests only? Or, are they meant to be used by other modules?

Reply:
Right now there's no centralized agent connection, so if other client modules ever need to query the agent, they'll need the socket path.

Reply:
dc_agent_sockpath seems to be exported for that purpose, no?

Reply:
Ah, you're right, not sure what I was thinking there. Yeah, the macros themselves could be private.